tests/infer_result_type_via_redefined/infer_result_type_via_redefined.fz

```diff
@@ -0,0 +1,35 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test infer_result_type_via_redefined
+#
+# -----------------------------------------------------------------------
+
+infer_result_type_via_redefined =>
+
```

Comment:
I am missing some test cases: 

- redefining two or more inherited features with equal types 
- redefining result types `a` and `b` as `void` 
- redefining `a.f unit`,  `b.f void`,  `c.f String` in 
  ```
    a is
      f unit =>
    b is
      f => do
    c is
      f => "hi"
    d : a, b, c is
      redef f => panic "bla"
  ```
  should IMHO work. 
 - negative tests